java/src/com/google/idea/blaze/java/run/hotswap/BlazeReloadFileAction.java

```diff
@@ -98,9 +106,22 @@ public void actionPerformed(@NotNull AnActionEvent anActionEvent) {
                     ProgressManager.getInstance().run(new Task.Backgroundable(project, "Preparing to hotswap", true) {
                         @Override
                         public void run(@NotNull ProgressIndicator progressIndicator) {
+                            // We want to be aware of entries like darwin_arm64-fastbuild/bin/some-api/lib-api.jar and not interested in darwin_arm64-fastbuild/external-api-0.0.654.jar
+                            // TODO: It is a good idea to skip non-output artifacts.
+                            List<String> artifactsForLogging = outputs.artifacts.keySet().stream().filter(artifact -> artifact.split("/").length > 2).collect(Collectors.toList());
```

Comment:
indexOf or contains should be more performant

Reply:
Maybe make this lazy so its only evaluated if we need to log it

Reply:
I want to make sure that there is only one `/` there so not sure how `contains` or `indexOf` can help here...

Reply:
Your logic currently checks `artifact.split("/").length > 2` meaning at least one `/`

Reply:
it should be more than 2 though... reworked to make sure that more than one